pkg/apis/app/v1alpha1/kogitoservices_types.go

```diff
@@ -169,6 +169,23 @@ func (k *KogitoServiceSpec) AddEnvironmentVariable(name, value string) {
 	return
 }
 
+// AddEnvironmentVariableFromSecret adds a new environment variable from the secret under the key
+func (k *KogitoServiceSpec) AddEnvironmentVariableFromSecret(variableName, secretName, secretKey string) {
```

Comment:
maybe add a unit test for it?
plus, isn't a similar func available somewhere else?

Reply:
There is a similar function but for a different purpose (setting variable directly on the container):
https://github.com/kiegroup/kogito-cloud-operator/blob/d6ed53bbeccf739a14efa51dfb658d4b69eb72db/pkg/framework/env_var.go#L110-L129

Also a test would be maybe helpful, although right now there isn't any test class for it so maybe we can address it as a separate JIRA if it is really needed? But to be honest, this function is really short and trivial so probably I wouldn't waste time with writing a test for it right now.